### ISSUE TITLE
Invalid value is assigned to variable in case if value contains 2 other variables in Step 'When I execute steps while counter...'

### ISSUE THREAD

**Issue by NadzeyaDunayeva** [vividus-framework/vividus#701]
**STR:**
1. Run following scenario:
```
When I execute SQL query `SELECT * FROM vividus` against `vividus` and save result to scenario variable `first`
When I execute steps while counter is not_equal_to `2` with increment `1` starting from `0`:
|step|
|When I initialize the STORY variable `qual_#{eval(${iterationVariable}+1)}` with value `${first[${iterationVariable}].id}`|
|When I initialize the STORY variable `qual_#{eval(${iterationVariable}+1)}` with value `${first[0].id}`|
```
2. Observe results

**Actual Result:**
step When I initialize the STORY variable `qual_#{eval(${iterationVariable}+1)}` with value `${first[${iterationVariable}].id}` returns for example
Saving a value '${first[0].id}' into the 'STORY' variable 'qual_1' 
step When I initialize the STORY variable `qual_#{eval(${iterationVariable}+1)}` with value `${first[0].id}` returns for example
Saving a value '1' into the 'STORY' variable 'qual_1' 
**Expected Result:**
in both cases it is expected to see Saving a value '1' into the 'STORY' variable 'qual_1' 

**Artifacts**
If applicable, add artifacts to help explain your problem:
 - Allure report [allure vividus issue 1.zip](https://github.com/vividus-framework/vividus/files/4837422/allure.vividus.issue.1.zip)
 - Test execution logs [vividus issue 1.log](https://github.com/vividus-framework/vividus/files/4837429/vividus.issue.1.log)

**Execution environment (please complete the following information):**
vividus: 0.2.4-SNAPSHOT

**Additional context**
In case if there is range of values and there is necessity to set variables with each value from range, there is no ability to loop over all values in range using dynamic index

**Comment by ikalinin1** [comment]
duplicates #691

**Comment by NadzeyaDunayeva** [comment]
Issue is fixed in new version of vividus: 0.2.4-SNAPSHOT. Closing
